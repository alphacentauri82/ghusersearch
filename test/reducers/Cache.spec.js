import cacheReducer, {addToCache} from 'reducers/Cache';

describe('Reducer: cache', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        cacheReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('addToCache function', () => {

    describe('when a cache entry does not exist', () => {
      it('should add an entry to the cache', () => {
        const state = {
          foo: {},
        };
        const newState = addToCache({
          state,
          cacheKey: 'thing',
          type: 'foo',
          data: {bar: 'baz'},
        });
        expect(newState).toMatchSnapshot();
      });
    });

    describe('when the cacheKey is undefined', () => {
      it('should just return the state', () => {
        const state = {
          foo: {
            exists: 'hi',
          },
        };
        const newState = addToCache({
          state,
          cacheKey: undefined,
          type: 'foo',
          data: {bar: 'baz'},
        });
        expect(newState).toMatchSnapshot();
      });
    });

    describe('when a cache entry does exist', () => {
      it('should just return the state', () => {
        const state = {
          foo: {
            exists: 'hi',
          },
        };
        const newState = addToCache({
          state,
          cacheKey: 'exists',
          type: 'foo',
          data: {bar: 'baz'},
        });
        expect(newState).toMatchSnapshot();
      });
    });

  });

  describe('followers cache', () => {
    it('should add an entry to the cache', () => {
      const beforeState = {
        followers: {},
      };
      const action = {
        type: 'FOLLOWERS_SUCCESS',
        payload: {
          entities: 'ignore',
          result: [1, 2],
          url: 'followers.net',
        },
      };
      expect(cacheReducer(beforeState, action)).toMatchSnapshot();
    });
  });

  describe('repos cache', () => {
    it('should add an entry to the cache', () => {
      const beforeState = {
        repos: {},
      };
      const action = {
        type: 'REPOS_SUCCESS',
        payload: {
          entities: 'ignore',
          result: [1, 2],
          url: 'repos.net',
        },
      };
      expect(cacheReducer(beforeState, action)).toMatchSnapshot();
    });
  });

  describe('search cache', () => {
    it('should add an entry to the cache', () => {
      const beforeState = {
        search: {},
      };
      const action = {
        type: 'SEARCH_SUCCESS',
        payload: {
          entities: 'ignore',
          search: 'page=2',
          result: [1, 2],
          totalResults: 2,
        },
      };
      expect(cacheReducer(beforeState, action)).toMatchSnapshot();
    });
  });

  describe('profile cache', () => {
    it('should add an entry to the cache', () => {
      const beforeState = {
        profile: {},
      };

      const action = {
        type: 'PROFILE_SUCCESS',
        payload: {
          login: 'simonsmith',
          name: 'Simon Smith',
        },
      };
      expect(cacheReducer(beforeState, action)).toMatchSnapshot();
    });
  });

});
