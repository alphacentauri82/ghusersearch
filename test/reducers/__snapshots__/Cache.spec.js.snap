// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: cache addToCache function when a cache entry does exist should just return the state 1`] = `
Object {
  "foo": Object {
    "exists": "hi",
  },
}
`;

exports[`Reducer: cache addToCache function when a cache entry does not exist should add an entry to the cache 1`] = `
Object {
  "foo": Object {
    "thing": Object {
      "bar": "baz",
    },
  },
}
`;

exports[`Reducer: cache addToCache function when the cacheKey is undefined should just return the state 1`] = `
Object {
  "foo": Object {
    "exists": "hi",
  },
}
`;

exports[`Reducer: cache followers cache should add an entry to the cache 1`] = `
Object {
  "followers": Object {
    "followers.net": Object {
      "result": Array [
        1,
        2,
      ],
    },
  },
}
`;

exports[`Reducer: cache profile cache should add an entry to the cache 1`] = `
Object {
  "profile": Object {
    "simonsmith": Object {
      "login": "simonsmith",
      "name": "Simon Smith",
    },
  },
}
`;

exports[`Reducer: cache repos cache should add an entry to the cache 1`] = `
Object {
  "repos": Object {
    "repos.net": Object {
      "result": Array [
        1,
        2,
      ],
    },
  },
}
`;

exports[`Reducer: cache search cache should add an entry to the cache 1`] = `
Object {
  "search": Object {
    "page=2": Object {
      "result": Array [
        1,
        2,
      ],
      "search": "page=2",
      "totalResults": 2,
    },
  },
}
`;

exports[`Reducer: cache when passed no state should return the initial state 1`] = `
Object {
  "followers": Object {},
  "profile": Object {},
  "rateLimit": Object {},
  "repos": Object {},
  "search": Object {},
}
`;
